lib/java/collections.nit

```diff
@@ -91,13 +133,47 @@ extern class JavaArray in "Java" `{ java.lang.Object[] `}
 	redef fun length in "Java" `{ return recv.length; `}
 end
 
-# TODO other primitive arrays:
-# * Java primitive array `byte[]`
-# * Java primitive array `short[]`
-# * Java primitive array `int[]`
-# * Java primitive array `long[]`
-# * Java primitive array `boolean[]`
-# * Java primitive array `char[]`
+# Java primitive array `boolean[]`
+extern class JavaBoolArray in "Java" `{ boolean[] `}
+	super AbstractJavaArray[Bool]
+
+	# Get a new array of the given `size`
+	new (size: Int) in "Java" `{ return new boolean[(int)size]; `}
+
+	redef fun [](i) in "Java" `{ return (boolean)recv[(int)i]; `}
+
+        redef fun []=(i, e) in "Java" `{recv[(int)i] = (boolean)e; `}
+
+	redef fun length in "Java" `{ return recv.length; `}
+end
+
+# Java primitive array `byte[]`
+extern class JavaByteArray in "Java" `{ byte[] `}
+	super AbstractJavaArray[Int]
+
+	# Get a new array of the given `size`
+        new (size: Int) in "Java" `{ return new byte[(int)size]; `}
+
+        redef fun [](i) in "Java" `{ return (byte)recv[(int)i]; `}
+
+        redef fun []=(i, e) in "Java" `{ recv[(int)i] = (byte)e; `}
+
+        redef fun length in "Java" `{ return recv.length; `}
+end
+
+# Java primitive array `char[]`
+extern class JavaCharArray in "Java" `{ char[] `}
+	super AbstractJavaArray[Char]
+
+	# Get a new array of the given `size`
+        new (size: Int) in "Java" `{ return new char[(int)size]; `}
+
+        redef fun [](i) in "Java" `{ return (char)recv[(int)i]; `}
```

Comment:
We'll need to be careful with these too, with Java's surrogate pairs, we can be surprised with this operation too as an access can yield an invalid code point in Nit :/